### ISSUE TITLE
[HOWTO] install org-ref in doom-emaxy way

### ISSUE THREAD

**Issue by zenny** [hlissner/doom-emacs#1573]
Hi,

Great project.

I tried to install a wonderful package https://github.com/jkitchin/org-ref into doom-emacs.

But there is no package with M-x package-install.

I added a line in .emacs.d/init.el it asks me to install epdinfo , when confirmed with an yes, sudo password was asked, but reported the specific letter of the password was UNDEFINED after pressing each key of the letter!

M-x doom/info output is as follows:

```
- OS: gnu/linux (x86_64-pc-linux-gnu)
- Shell: /bin/bash
- Emacs: 26.2 (Jun 03, 2019)
- Doom: 2.0.9 (HEAD -> develop, origin/develop, origin/HEAD c99c184 2019-07-09 23:30:59 +0200)
- Graphic display: t (daemon: nil)
- System features: XAW3D XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS LUCID X11 XDBE XIM THREADS LCMS2
- Details:
  ```elisp
  env bootstrapper: envvar-file
  elc count: 0
  uname -a:  Linux 4.4.0-148-lowlatency #174~14.04.1-Ubuntu SMP PREEMPT Thu May 9 08:50:56 UTC 2019 x86_64
  modules:   (:completion company ivy :ui doom doom-dashboard doom-quit hl-todo modeline nav-flash ophints (popup +all +defaults) treemacs vc-gutter vi-tilde-fringe window-select workspaces :editor evil file-templates fold multiple-cursors rotate-text snippets :emacs dired electric vc :term eshell term vterm :tools ansible eval flycheck (lookup +docsets) magit pdf tmux :lang cc data erlang elixir elm emacs-lisp go javascript latex markdown (org +attach +babel +capture +export +habit +present +protocol +ref) perl php python sh web :app write :config default)
  packages:  (org-ref-pdf-directory org-ref-bibliography-notes org-ref-default-bibliography)
  exec-path: (~/.asdf/shims ~/.asdf/bin /mnt/homepool/HOME/zenny/Downloads/mono2colour/torch/install/bin . /mnt/homepool/HOME/zenny/Downloads/mono2colour/torch/install/bin . /mnt/homepool/HOME/zenny/Downloads/mono2colour/torch/install/bin ~/.local/share/umake/bin ~/bin /usr/local/sbin /usr/local/bin /usr/sbin /usr/bin /sbin /bin /usr/games /usr/local/games /usr/brlcad/bin /usr/lib/jvm/java-8-oracle/bin /usr/lib/jvm/java-8-oracle/db/bin /usr/lib/jvm/java-8-oracle/jre/bin ~/.fzf/bin /usr/local/libexec/emacs/26.2/x86_64-pc-linux-gnu/)
  ```
What is the ideal way to install a new package in doom-emacs? Can anyone elaborate?

Cheers,
/z

**Comment by netpro2k** [comment]
Emacs/doom noob myself, but the "doom way" to add new packages is to add them to your private module (`~/.doom.d` directory). You can find more info in the wiki here: https://github.com/hlissner/doom-emacs/wiki/Customization#install-my-own-packages and here https://github.com/hlissner/doom-emacs/wiki/Modules

tldr, something like (didn't try this myself):

In `~/.doom.d/packages.el`:

```elisp
(package! org-ref)
```

In `~/.doom.d/config.el`:

```elisp
(def-package! org-ref
    :after org
    :init
    ; code to run before loading org-ref
    :config
    ; code to run after loading org-ref
    )
```
Then run `~/.emacs.d/bin/doom refresh`

**Comment by zenny** [comment]
> Emacs/doom noob myself, but the "doom way" to add new packages is to add them to your private module (`~/.doom.d` directory). You can find more info in the wiki here: https://github.com/hlissner/doom-emacs/wiki/Customization#install-my-own-packages and here https://github.com/hlissner/doom-emacs/wiki/Modules
> 
> tldr, something like (didn't try this myself):
> 
> In `~/.doom.d/packages.el`:
> 
> ```emacs-lisp
> (package! org-ref)
> ```
> 
> In `~/.doom.d/config.el`:
> 
> ```emacs-lisp
> (def-package! org-ref
>     :after org
>     :init
>     ; code to run before loading org-ref
>     :config
>     ; code to run after loading org-ref
>     )
> ```
> 
> Then run `~/.emacs.d/bin/doom refresh`

@netpro2k Thanks for your input, shall try.

**Comment by ocharles** [comment]
This doesn't seem to work for me:

```
    → Cloning org-ref...
    → Cloning org-ref...done
    → Building org-ref...
    → Building org-ref → Cloning helm...
    → Building org-ref → Cloning helm...done
    → Building org-ref → Building helm...
    → Building org-ref → Building helm → Building helm-core...
    → Building org-ref → Building helm → Building helm-core...done
    → Building org-ref → Building helm...
    → Building org-ref → Building helm...done
    → Building org-ref → Cloning helm-bibtex...
    → Building org-ref → Cloning helm-bibtex...done
    → Building org-ref → Building helm-bibtex...
x There was an unexpected error:
  Type: doom-package-error
  Message:
    Error with packages
  Data:
    "org-ref"
    (error "Could not find package bibtex-completion in recipe repositories: (org-elpa melpa gnu-elpa-mirror emacsmirror-mirror)")
    ""
```

I'm not quite sure what to do about this.

**Comment by hlissner** [comment]
@ocharles Unrelated issue. See https://github.com/hlissner/doom-emacs/issues/2802#issuecomment-606654151.

Since this issue was answered I'll consider it resolved.
